File: 0371-sum-of-two-integers/0371-sum-of-two-integers.cpp
class Solution {
public:
    int getSum(int a, int b) {
        while(b!=0){
            //jab carry not equal to 0 na ho jo
            //update b ma hoga
            int carry=a&b;//carry1
            a=a^b;//without carry add
            b=carry<<1;//update carry
        }
        return a;
    }
};

File: 0371-sum-of-two-integers/0371-sum-of-two-integers.js
/**
 * @param {number} a
 * @param {number} b
 * @return {number}
 */
var getSum = function(a, b) {
    while(b!=0){
        let carry=a&b;//carry
        a=a^b;//without carry
        b=carry<<1;
    }
    return a;
};